### ISSUE TITLE
Question about the status of WASI-Poll in Preview 3

### ISSUE THREAD

**Issue by rsta79** [WebAssembly/wasi-poll#16]
Hey there,

I have a question about the status of WASI Poll in Preview 3. I heard that the resource feature isn't ready for Preview 2 yet, and that WASI proposals are using u32 types as a temporary workaround to represent indices into a global table of handles. With that in mind, I was wondering if you could tell me whether WASI Poll will be deprecated, kept for compatibility with Preview 2, or integrated into the asynchronous model in Preview 3. I'd really appreciate any insights you can share on this topic.

**Comment by sunfishcode** [comment]
The current wasi-poll uses the u32 indices with the implied global table. This is temporary scaffolding meant to work as much like resources and handles as possible, to unblock development on other things in WASI.

Resources are under active development. When they're ready, wasi-poll and all the other WASI APIs will quickly switch to use them. This is expected to happen before Preview2 is stabilized. wasi-poll will still be needed, because it's the only polling mechanism.

Preview3 will add `future` and `stream` types to the type system, and along with them it is expected that Preview3 will provide a built-in polling mechanism. At that time, wasi-poll is expected to be redundant. We may keep it for some time for compatibility with code using it, but it may be deprecated and removed when no longer needed.

**Comment by rsta79** [comment]
Thanks for the quick reply, @sunfishcode! Your response addressed all of my questions, and I will now close this issue. Hopefully, this will be helpful for someone who has the same questions!
